### DISCUSSION TITLE
Allowing zero-like values as env variables

### DISCUSSION THREAD

**Discussion by bryan-garcia** [llnl/maestrowf#390] [Q&A]
Hi Everyone,

There is an issue when providing integer 0 or float 0 in the env.variables block that triggers a ValueError being raised. Specifically it's in [yamlspecification.py](https://github.com/LLNL/maestrowf/blob/develop/maestrowf/specification/yamlspecification.py) (method _verify_variables on line 190):
```python
if not value:
    msg = (
        "All variables must have a valid value. Empty strings "
        "are not allowed."
    )
    logger.error(msg)
    raise ValueError(msg)
```

A value of integer/float 0 in the env.variables block clearly triggers this error.

Sometimes I don't want to use the global.parameter block to define my values, and simply want an env variable. What are some thoughts for allowing this from a design perspective?

**Comment by FrankD412** [comment]
Hi @bryan-garcia -- So this looks like it was a bug. Were you thinking of anything other than having zero value numerical values in the environment block? Are there other use cases you were thinking of?

If it was just the bug, I can close this conversation since it's a duplicate of #391 and is fixed by #393 (if this was solely a bug).

**Reply by bryan-garcia** [reply]
Hi Frank,

I think we can close this discussion. Thanks for addressing this!

**Reply by FrankD412** [reply]
No problem! -- happy to help 😄

**Comment by FrankD412** [comment]
Handled by #393
